extensions/src/main/java/org/pentaho/platform/plugin/services/webservices/content/AbstractAxisServiceContentGenerator.java

```diff
@@ -117,10 +117,28 @@ protected void processAxisFault( MessageContext msgContext, OutputStream out, Th
       } else {
         // set the status of the HTTP response
         String status = (String) msgContext.getProperty( Constants.HTTP_RESPONSE_STATE );
-        if ( status == null ) {
-          res.setStatus( HttpServletResponse.SC_INTERNAL_SERVER_ERROR );
-        } else {
-          res.setStatus( Integer.parseInt( status ) );
+        try {
+          if ( status == null ) {
+            res.sendError( HttpServletResponse.SC_INTERNAL_SERVER_ERROR );
+          } else {
+            int sc = Integer.parseInt( status );
+            if ( sc >= 400 ) {
+              res.sendError( sc );
+            } else {
+              res.setStatus( sc );
+            }
+          }
+        } catch ( IOException e1 ) {
+          if ( status == null ) {
+            res.setStatus( HttpServletResponse.SC_INTERNAL_SERVER_ERROR );
+          } else {
+            int sc = Integer.parseInt( status );
+            if ( sc >= 400 ) {
```

Comment:
Don't think you need this if - it's doing the same thing in the if and the else.

Reply:
Yes thanks it is mistype - will change